src/pages/RFOX/hooks/useEpochHistoryQuery.ts

```diff
@@ -32,12 +34,18 @@ export const epochHistoryQueryFn = async (): Promise<EpochMetadata[]> => {
 
   while (startTimestamp < now) {
     const nextStartTimestamp = BigInt(dayjs.unix(Number(startTimestamp)).add(1, 'month').unix())
+    startTimestamp = nextStartTimestamp
```

Comment:
Checks-effects-interactions of sorts, ensure we don't go infinite loop if the block below is evaluated and we break